src/main/java/bio/terra/service/dataset/DatasetDao.java

```diff
@@ -512,13 +512,16 @@ public DatasetSummary retrieveSummaryByName(String name) {
 
         // add the filter to the clause to get the actual items
         DaoUtils.addFilterClause(filter, params, whereClauses);
+
+        // results in 3x the dataset records
+        String joinSql = "LEFT JOIN storage_resource WHERE storage_resource.dataset_id = dataset.id";
```

Comment:
I think you could optionally add this to the where clause rather than a left join.  In that case, it could be something like, in the `addAuthzIdsClause` function (might need a separate function since this isn't really authz):
```
if (there's a region filter) {
    clauses.add("exists (select 1 from storage_resource WHERE storage_resource.dataset_id = dataset.id)")
}
```
This way, you row counts stay the same

Reply:
We also need to check if the region matches, so could something like this work? 
` "filter in (select storage_resource.region from storage_resource WHERE storage_resource.dataset_id = dataset.id)"`